test/functional/JLM_Tests/src/org/openj9/test/java/lang/management/TestOpenJ9DiagnosticsMXBean.java

```diff
@@ -414,15 +420,25 @@ private void triggerDumpToFile(OpenJ9DiagnosticsMXBean diagBean, String test) th
 				}
 				String fileName = diagBean.triggerDumpToFile(agent, filePath);
 				File dumpFile = new File(fileName);
-
 				if (!(agent.equals("stack"))) {
 					if (isZOSSystemAgent(agent)) {
-						/* checking for dataset presence will be included later */
+						String actual_dsn = "";
+						logger.info("Dataset name returned by triggerDumpToFile = " + fileName);
+						if (fileName.endsWith("X&DS")) {
+							logger.info("Replacing X&DS with X001 in system dump dataset name");
+							actual_dsn = fileName.replace("X&DS", "X001");
```

Comment:
This should replace the suffix not just any occurrence of `X&DS`.
The first argument to `replace` is a regular expression which might misinterpret `&`.

Reply:
I don't see the dumps getting created are always mult-ipart or being suffixed with X&DS, hence included a check to detect if fileName ends with "X&DS" and if so replace with "X001"

http://vmfarm.ottawa.ibm.com/job_output.php?id=547308220
[TestOpenJ9DiagnosticsMXBean] [INFO] Dataset name returned by triggerDumpToFile = J9BUILD.MYJVM.J9BUILD1.D180530.T030804
[TestOpenJ9DiagnosticsMXBean] [INFO] Dataset name does not contain X&DS

http://vmfarm.ottawa.ibm.com/job_output.php?id=547307586
[TestOpenJ9DiagnosticsMXBean] [INFO] Dataset name returned by triggerDumpToFile = J9BUILD.MYJVM.J9BUILD5.D180530.T040442.X&DS
[TestOpenJ9DiagnosticsMXBean] [INFO] Replacing X&DS with X001 in system dump dataset name

Reply:
My point is that the condition uses `endsWith()` so the adjustment should only replace an occurrence of that string at the *end*.